lib/matplotlib/backend_bases.py

```diff
@@ -3195,3 +3205,474 @@ def zoom(self, *args):
     def set_history_buttons(self):
         """Enable or disable back/forward button"""
         pass
+
+
+class ToolEvent(object):
```

Comment:
@fariza If these comments are redundent/outdated may I delete them?  They seem to have become de-coupled from the code, but I don't want to delete your messages with you your consent and I don't want to delete mine with out deleting yours (as they will end up being strangely orphaned).